test-app/runtime/src/main/cpp/Runtime.cpp

```diff
@@ -56,11 +60,15 @@ void Runtime::Init(JavaVM* vm, void* reserved) {
         JEnv::Init(s_jvm);
     }
 
+    // handle SIGABRT/SIGSEGV only on API level > 25 as the handling is not so efficient in older versions
     if (m_androidVersion > 25) {
-        // handle SIGABRT only on API level > 25 as the handling is not so efficient in older versions
-        struct sigaction action;
-        action.sa_handler = SIGABRT_handler;
-        sigaction(SIGABRT, &action, NULL);
+        struct sigaction SIGABRT_action;
+        SIGABRT_action.sa_handler = SIGABRT_handler;
+        sigaction(SIGABRT, &SIGABRT_action, NULL);
+
+        struct sigaction SIGSEGV_action;
+        SIGSEGV_action.sa_handler = SIGSEGV_handler;
```

Comment:
We can reuse the same signal handler for both signals, since the behaviour is quite similar (throw a `NativeScriptException`).